src/compiler/moduleNameResolver.ts

```diff
@@ -1808,6 +1808,10 @@ namespace ts {
     /*@internal*/
     export interface PackageJsonInfo {
         packageDirectory: string;
+        contents: PackageJsonInfoContents;
+    }
+    /*@internal*/
+    export interface PackageJsonInfoContents {

```

Comment:
I guess we don't have so many package.json files that the extra allocations will be a problem?